app/assets/javascripts/components/CartedProducts.jsx

```diff
@@ -0,0 +1,65 @@
+var CartedProducts = React.createClass({
+  getInitialState: function(){
+     return {
+       initialItems: this.props.carted_products,
+       carted_products: []
+     }
+  },
+  componentWillMount: function(){
+    this.setState({carted_products: this.state.initialItems})
+  },
+  updateQuantity: function(val, id) {
+    console.log(val, "from parent && id: ", id);
+    this.props.carted_products.map((carted_product) =>
+      (carted_product.id == id) ? (
+        carted_product.quantity = val
+      ) : (
+        carted_product
+      )
+    );
+    $.ajax({
```

Comment:
@DillonStreator just to follow up...I know I mentioned we should try and not use jquery...but after doing some digging...it really is the easiest way to make this ajax call. We are kind of limited inside of our rails app with how we've integrated react. I thought this was an interesting article: [5 best libraries for making AJAX calls in React](https://hashnode.com/post/5-best-libraries-for-making-ajax-calls-in-react-cis8x5f7k0jl7th53z68s41k1).